sdk/cosmos/src/clients/cosmos_client.rs

```diff
@@ -83,20 +111,58 @@ impl CosmosClient {
             uri,
         };
         Self {
-            http_client,
+            transport: TransportStack::Client(http_client),
             auth_token,
             cloud_location,
         }
     }
 
+    /// TODO
+    pub fn with_pipeline(
+        account: String, // TODO: this will eventually be a URL
+        auth_token: AuthorizationToken,
```

Comment:
At some point, this should be `credential` of type `TokenCredential`. We have a whole pluggable system on this. Some `TokenCredential` providers just wrap a field and supply that to a policy, like shared key credentials (api-key, etc.). See https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/AzureKeyCredential.cs for an example.

Reply:
Yes good point. Right now the authorization is tightly coupled with the client. We should extract the code and put it in a specific policy.